util/ranger/points.go

```diff
@@ -205,18 +208,41 @@ func (r *builder) buildFormBinOp(expr *expression.ScalarFunction) []point {
 		err   error
 		ft    *types.FieldType
 	)
+	_, collation, _ := expr.CharsetAndCollation(expr.GetCtx())
 	if col, ok := expr.GetArgs()[0].(*expression.Column); ok {
```

Comment:
I  think we have better put these two branches into a function, their logics are quite similar.